src/main/java/be/seeseemelk/mockbukkit/entity/BoatMock.java

```diff
@@ -0,0 +1,171 @@
+package be.seeseemelk.mockbukkit.entity;
+
+import be.seeseemelk.mockbukkit.ServerMock;
+import be.seeseemelk.mockbukkit.UnimplementedOperationException;
+import com.google.common.base.Preconditions;
+import org.bukkit.Material;
+import org.bukkit.TreeSpecies;
+import org.bukkit.entity.Boat;
+import org.bukkit.entity.EntityType;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.UUID;
+
+public class BoatMock extends VehicleMock implements Boat
+{
+
+	private @NotNull Type boatType = Type.OAK;
+	private double maxSpeed = 0.4D;
+	private double ocupiedDeceleration = 0.2D;
+	private double unoccupiedDeceleration = -1;
+	private boolean workOnLand = false;
+
+	/**
+	 * Constructs a new {@link BoatMock} on the provided {@link ServerMock} with a specified {@link UUID}.
+	 *
+	 * @param server The server to create the entity on.
+	 * @param uuid   The UUID of the entity.
+	 */
+	public BoatMock(@NotNull ServerMock server, @NotNull UUID uuid)
+	{
+		super(server, uuid);
+	}
+
+	@Override
+	@Deprecated
+	public @NotNull TreeSpecies getWoodType()
+	{
+		return switch (boatType)
+		{
+			case SPRUCE -> TreeSpecies.REDWOOD;
+			case BIRCH -> TreeSpecies.BIRCH;
+			case JUNGLE -> TreeSpecies.JUNGLE;
+			case ACACIA -> TreeSpecies.ACACIA;
+			case DARK_OAK -> TreeSpecies.DARK_OAK;
+			default -> TreeSpecies.GENERIC;
+		};
+	}
+
+	@Override
+	@Deprecated
+	public void setWoodType(@NotNull TreeSpecies species)
+	{
+		Preconditions.checkNotNull(species, "Species can not be null");
+		Type type;
+
+		type = switch (species)
+		{
+			case GENERIC -> Type.OAK;
+			case BIRCH -> Type.BIRCH;
+			case ACACIA -> Type.ACACIA;
+			case JUNGLE -> Type.JUNGLE;
+			case DARK_OAK -> Type.DARK_OAK;
+			case REDWOOD -> Type.SPRUCE;
+		};
+
+		this.boatType = type;
+
+	}
+
+	@Override
+	public @NotNull Type getBoatType()
+	{
+		return this.boatType;
+	}
+
+	@Override
+	public void setBoatType(@NotNull Type type)
+	{
+		Preconditions.checkNotNull(type, "Type cannot be null");
+		this.boatType = type;
+	}
+
+	@Override
+	@Deprecated
+	public double getMaxSpeed()
+	{
+		return this.maxSpeed;
+	}
+
+	@Override
+	@Deprecated
+	public void setMaxSpeed(double speed)
+	{
+		Preconditions.checkArgument(speed >= 0.0D, "Speed cannot be negative");
```

Comment:
Looks like this silently fails in paper; if the speed is smaller than 0, the request to change it is ignored

Reply:
I think it's fair to fail here. If it silently fails, the value shouldn't be less than 0, so we should alert the developer to that. Code that runs on a server doesn't *have* to run on MockBukkit, but code that runs on MockBukkit should run on a server.

Reply:
Good point, then I think it's fine as is